packages/ubuntu_wsl_setup/lib/pages/profile_setup/profile_setup_model.dart

```diff
@@ -48,9 +60,10 @@ class ProfileSetupModel extends ChangeNotifier {
 
   /// Whether the current input is valid.
   bool get isValid =>
-      RegExp(kValidUsernamePattern).hasMatch(username) &&
+      realname.isNotEmpty &&
       password.isNotEmpty &&
-      password == confirmedPassword;
+      password == confirmedPassword &&
+      RegExp(kValidUsernamePattern).hasMatch(username);
```

Comment:
nitpick: I would keep the check in the same here of the attribute appearance in both UIs (realname/username/password) or in ascii order.

Reply:
sorry, the reason i tried slipping this change in was to have the slowest regexp check as last

Reply:
no worry, was just wondering why :)